sycl/include/CL/sycl/detail/assert_happened.hpp

```diff
@@ -0,0 +1,40 @@
+//==------- assert_happened.hpp - Assert signalling structure --------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/detail/defines_elementary.hpp>
+
+#ifdef __SYCL_DEVICE_ONLY__
+// Reads Flag of AssertHappened on device
+SYCL_EXTERNAL __attribute__((weak)) extern "C" void
+__devicelib_assert_read(void *);
+#endif
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+struct AssertHappened {
```

Comment:
Suggest documenting layout of this structure in the doc.

Reply:
Fixed.